### PR TITLE
Use Apollo instead of our custom GraphQL hooks for the diagram part

### PR DESCRIPTION
**Pull request by pcdavid**
This is not completely finished, in particular I have not tested the error handling. In particular the `connection_error` message type which was handled before is ignored at the moment. I'm publishing it anyway to get feedback on the rest while I continue on the error handling.

### CODE REVIEW

**Review by sbegaudeau — COMMENTED** [review]
I think that pretty much everything in this file is wrong. You are adding more of the same "wrongdoing" to this file which may work but we will have to fix that.

Neither or dev dependencies nor our peer dependencies make sense.

**Review by sbegaudeau — COMMENTED** [review]
I agree with this code but I wonder if it should not be in sirius-web instead.

For example, I don't see why we should have anything to do with credentials support. Especially since different users of `sirius-components` already have different authentication strategies.

File: frontend/package.json
Comment (sbegaudeau) [inline frontend/package.json:28]: I think that pretty much everything in this file is wrong. You are adding more of the same "wrongdoing" to this file which may work but we will have to fix that.

Neither or dev dependencies nor our peer dependencies make sense.

File: frontend/src/common/ApolloGraphQLClient.ts
Comment (sbegaudeau) [inline frontend/src/common/ApolloGraphQLClient.ts:1]: I agree with this code but I wonder if it should not be in sirius-web instead.

For example, I don't see why we should have anything to do with credentials support. Especially since different users of `sirius-components` already have different authentication strategies.

File: frontend/src/diagram/DiagramWebSocketContainer.tsx
Comment (sbegaudeau) [inline frontend/src/diagram/DiagramWebSocketContainer.tsx:421]: The reducer should be modified to use what it needs from the server response instead of just modifying it inline.

JSON.parse + JSON.stringify is overkill to not use some basic rest and spread.

File: frontend/src/diagram/reducer.ts
Comment (sbegaudeau) [inline frontend/src/diagram/reducer.ts:211]: No unit tests to update?

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: frontend/src/diagram/reducer.ts
Comment (pcdavid) [reply frontend/src/diagram/reducer.ts:211]: Fixed by https://github.com/eclipse-sirius/sirius-components/pull/128/commits/0f6add0edf420bcbba4c3a325b7927a3c182746f

File: frontend/src/diagram/DiagramWebSocketContainer.tsx
Comment (pcdavid) [reply frontend/src/diagram/DiagramWebSocketContainer.tsx]: Fixed by https://github.com/eclipse-sirius/sirius-components/pull/128/commits/44ce7aede446bb4bced5cd8e72c5bb2d66911ef3

File: frontend/package.json
Comment (pcdavid) [reply frontend/package.json:28]: Agreed, but if you're OK we'll handle this issue in a separate PR. It's not worse than what was done before, so I don't think this should block merging this as long as we plan to fix the global issue soon.

File: frontend/src/common/ApolloGraphQLClient.ts
Comment (pcdavid) [reply frontend/src/common/ApolloGraphQLClient.ts:1]: I understand the concern, but this would lead to duplicate code in all apps based on `sirius-components`. I'll see if we can tweak the `ApolloClient` after it has been constructed, so that we can provide here one with some default config, and apps that need it could add the `credentials` part if they need it.

Note that the current custom hooks already mix in some authentication-related code, e.g.:
* https://github.com/eclipse-sirius/sirius-components/blob/master/frontend/src/common/GraphQLHooks.ts#L185
* https://github.com/eclipse-sirius/sirius-components/blob/master/frontend/src/common/GraphQLHttpClient.ts#L26 (hard-coded `credentials: 'include',  mode: 'cors'`).

Comment (pcdavid) [reply frontend/src/common/ApolloGraphQLClient.ts]: Fixed by https://github.com/eclipse-sirius/sirius-components/pull/128/commits/4a118e75b505730d3251d0e596a894ce719a727e, and https://github.com/eclipse-sirius/sirius-web/pull/12/commits/8d298629cbd08798d4678fe0bedf158e4d732f6f on the sirius-web side.

**Comment by sbegaudeau** [comment]
This seems fine, I'll just wait for the fix for the latest details on https://github.com/eclipse-sirius/sirius-web/pull/12 to merge them both.

**Review by sbegaudeau — APPROVED** [review]

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
I'll open an issue in order to activate the squash and merge option but in the meantime since everything is now ok, could you squash the whole PR and then I'll merge it. Thanks

**Comment by sbegaudeau** [comment]
Forget about the last comment, I can do it myself. Since everything is now ok, I'll make sure that everything is merged regarding Apollo.

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**
